frontend/packages/operator-lifecycle-manager/src/components/operand/index.tsx

```diff
@@ -120,7 +120,11 @@ const getOperandStatus = (obj: K8sResourceKind): OperandStatusType => {
     };
   }
 
-  const trueConditions = conditions?.filter((c: K8sResourceCondition) => c.status === 'True');
+  const conditionsIsObject =
+    typeof conditions === 'object' && !Array.isArray(conditions) && conditions !== null;
+  const conditionsArray = conditionsIsObject ? [conditions] : conditions;
```

Comment:
The `conditionsArray` name can be misleading because it could be null (or even a string if `conditions` can be a string or other non-object type).

Reply:
Good point, @stleerh.  How do you feel about `formattedConditions` instead?  Still not great...

Reply:
/lgtm